app/web/src/components/EncodedUrl/ExternalScript.tsx

```diff
@@ -0,0 +1,126 @@
+import { useState } from 'react'
+import { useIdeContext, ideTypeNameMap } from 'src/helpers/hooks/useIdeContext'
+import OutBound from 'src/components/OutBound/OutBound'
+import { prepareEncodedUrl, makeExternalUrl } from './helpers'
+import { copyTextToClipboard } from 'src/helpers/clipboard'
+import { useRender } from 'src/components/IdeWrapper/useRender'
+import { toast } from '@redwoodjs/web/toast'
+
+const ExternalScript = () => {
+  const { state, thunkDispatch } = useIdeContext()
+  const handleRender = useRender()
+  const [rawUrl, setRawUrl] = useState('')
+  const [script, setScript] = useState('')
+  const [asyncState, setAsyncState] =
+    useState<'INIT' | 'SUCCESS' | 'ERROR' | 'LOADING'>('INIT')
+
+  const cadName = ideTypeNameMap[state.ideType]
+
+  const onPaste: React.ClipboardEventHandler<HTMLInputElement> = async ({
+    clipboardData,
+  }) => {
+    const url = clipboardData.getData('Text')
+    processUserUrl(url)
+  }
+  const onChange: React.ChangeEventHandler<HTMLInputElement> = async ({
+    target,
+  }) => setRawUrl(target.value)
+  const onKeyDown = async ({ key, target }) =>
+    key === 'Enter' && processUserUrl(target.value)
```

Comment:
When a user pastes a url it will immediately start processing it without having a "submit" button. I think this is appropriate since the whole point is to link to an external URL and no one will be typing those, you can type and hit enter though.